### ISSUE TITLE
OutputReportTabularMonthly_ResetMonthlyGathering fails

### ISSUE THREAD

**Issue by nmerket** [NREL/EnergyPlus#5294]
When I build `develop` and run `energyplus_tests` it gives me an assertion error on this test. When I pass in the flag to only run that one unit test `--gtest_filter="*OutputReportTabularMonthly_ResetMonthlyGathering`, which is what the CI machines do, it passes. Possibly something is missing from a `clear_state` run in a previous test. @JasonGlazer I'm assigning this to you because you wrote this test.

**Comment by mjwitte** [comment]
@nmerket It's possible that this has been fixed by @EnergyArchmage in #5291 or @rraustad in #5216

**Comment by EnergyArchmage** [comment]
Yes, I ran into that as well and went ahead and add clear_stuff in pr 5291.  Noel if you merge in that branch (https://github.com/NREL/EnergyPlus/tree/5255-EMS-dual-Temp-Node-Setpoints) into your branch. it should solve the problem. 

I don't really think we want issues for this sort of thing.

**Comment by JasonGlazer** [comment]
@EnergyArchmage Thanks for addressing this.

**Comment by EnergyArchmage** [comment]
No problem, But it isn't really addressed until that PR 5291 gets merged... hint hint

**Comment by jasondegraw** [comment]
I am getting failures in
- OutputReportTabularMonthly_ResetMonthlyGathering
- ConfirmResetBEPSGathering

#5291 fixes the first one, but not the second one. Are we sure we don't need an issue?

**Comment by EnergyArchmage** [comment]
humm, when the unit test unit test stops working is there a new issue with another unit test for the unit test unit test?

**Comment by jasondegraw** [comment]
@EnergyArchmage Yes, I believe so. Here's what it should look like:

```
ASSERT_TRUE(Brent_will_fix_it());
```

**Comment by jasondegraw** [comment]
Kidding aside, my "change the license text" branch is probably not the best place to fix a failing unit test and I don't have time for it anyway. I can either reopen this issue or put in a new one for ConfirmResetBEPSGathering...

**Comment by JasonGlazer** [comment]
I just reopened it. As a work around, just use --gtest_filter=-_ConfirmResetBEPSGathering_

**Comment by JasonGlazer** [comment]
@jasondegraw I cannot reproduce this by using the "develop" branch. Is it possible that you were using a different branch?

**Comment by mbadams5** [comment]
@JasonGlazer I just pulled down current develop and it is definitely there. You need to run all tests locally, not just `--gtest_filter=*OutputReportTabularMonthly_ResetMonthlyGathering*`.

It is due to [this static member variable](https://github.com/NREL/EnergyPlus/blob/ddcd0d2256264e2813015baad01e388d7ec9fb8e/src/EnergyPlus/OutputReportTabular.cc#L3586), which is causing an [assertion error here](https://github.com/NREL/EnergyPlus/blob/ddcd0d2256264e2813015baad01e388d7ec9fb8e/src/EnergyPlus/OutputReportTabular.cc#L3616) when a different unit test run is out of range of what the first unit test set MonthlyTablesNumColumns value.

**Comment by mjwitte** [comment]
@mbadams5 @JasonGlazer I'm reviewing and testing #5291 so I can slip the fix for this into there if you tell me what I need to do.

**Comment by mbadams5** [comment]
@mjwitte I think that PR already fixes this... at least it looks like it [here](https://github.com/NREL/EnergyPlus/pull/5291/files#diff-cfc84b456bd550d5729f5275fb755f77R3775). It will now reset that static member variable each unit test run.

**Comment by mjwitte** [comment]
@mbadams5 I merged in develop on the 5291 branch and the tests fail, maybe its a different problem.  Haven't dug any further yet.

```
[ RUN      ] OutputReportTabularTest.ConfirmResetBEPSGathering
..\..\..\..\tst\EnergyPlus\unit\OutputReportTabular.unit.cc(1056): error: Value
of: gatherEndUseBEPS( 1, endUseExteriorLights )
  Actual: -2.77679e+062
Expected: extLitUse * 3
Which is: 3.03
```

**Comment by JasonGlazer** [comment]
@mjwitte I believe that is the second item that was found earlier in this conversation. I am going to try to address it as soon as I can reproduce it.

**Comment by mjwitte** [comment]
@JasonGlazer You should be able to reproduce it with branch 5255-EMS-dual-Temp-Node-Setpoints
It's failing on the first EXPECT in TEST( OutputReportTabularTest, ConfirmResetBEPSGathering )

**Comment by jasondegraw** [comment]
I just rebuilt develop and ran

```
energyplus_tests.exe --gtest_filter=-*OutputReportTabularMonthly*
```

and got the result Mike gives above. If I run 

```
energyplus_tests.exe --gtest_filter=*ConfirmResetBEPSGathering*
```

it passes.

**Comment by mjwitte** [comment]
I think I've got it fixed now in #5291  Will push up shortly.  Just needed to move TEST( OutputReportTabularTest, ConfirmResetBEPSGathering ) into EnergyPlusFixture.

**Comment by mjwitte** [comment]
Closed via #5291
